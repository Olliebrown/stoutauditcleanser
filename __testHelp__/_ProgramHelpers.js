import fs from 'fs'
import path from 'path'

// Project specific imports
import { QUERIES } from '../src/domTraversal/queriesAndRegex.js'
import Program from '../src/Objects/Program.js'

export function InitProgramTest (htmlFileName) {
  // Read in the HTML page
  const baseHTML = fs.readFileSync(
    path.join('__tests__', 'data', 'ProgramTables', htmlFileName),
    'utf8'
  )

  // Load page from imported HTML
  document.documentElement.innerHTML = baseHTML.toString()

  // Extract program headers and make sure there's at least one
  const headerNodes = document.querySelectorAll(QUERIES.programHeader)
  if (headerNodes?.length < 1) {
    throw new Error('Unable to find at least one program')
  }

  // Create program object
  return new Program(headerNodes[0])
}
