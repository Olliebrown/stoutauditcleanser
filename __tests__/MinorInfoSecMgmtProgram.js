// Necessary when using ECMA modules
import { jest } from '@jest/globals'

// Help load the program HTML page
import { InitProgramTest } from '../__testHelp__/_ProgramHelpers.js'
// Disable jest mocking for fs module
jest.dontMock('fs')

// Mock console.warn to disable output
console.warn = jest.fn()

describe('Mathematics Minor Node', () => {
  // Inject clean version of HTML page and rebuild program before tests
  let program = null
  beforeAll(() => {
    // Load page from imported HTML
    program = InitProgramTest('MinorInfoSecMgmt.html')
  })

  // Clear out the HTML and program
  afterAll(() => {
    jest.resetModules()
    program = null
  })

  // Basic page structure tests
  test('should have a specific page title and specific program name', () => {
    // Check title
    const title = document.querySelector('title')
    expect(title).toBeTruthy()
    expect(title.textContent).toBe('Minor Info Security Management')

    // Check program name
    expect(program).toBeTruthy()
    expect(program.getName()).toBe('Minor: InfoSec Mgmt')
  })

  test('should have expected snapshot', () => {
    expect(program?.toJSON()).toMatchSnapshot()
  })
})
