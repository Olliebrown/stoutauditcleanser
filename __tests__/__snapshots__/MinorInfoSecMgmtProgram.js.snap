// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mathematics Minor Node should have expected snapshot 1`] = `
{
  "internalId": "RG-11154",
  "key": "minor__infosec_mgmt",
  "name": "Minor: InfoSec Mgmt",
  "satisfied": "INCOMPLETE",
  "subNodes": [
    {
      "description": "",
      "index": 1,
      "internalId": "",
      "isGenEd": false,
      "key": "minor__infosec_mgmt_minor",
      "name": "Minor: InfoSec Mgmt Minor",
      "nextIndex": 8,
      "satisfied": "INCOMPLETE",
      "subNodes": [],
      "units": {},
    },
  ],
  "units": {},
}
`;
